sonic_installer/bootloader/bootloader.py

```diff
@@ -0,0 +1,70 @@
+
+import os
+import re
+import subprocess
+
+from ..common import (
+   IMAGE_DIR_PREFIX,
+   IMAGE_PREFIX,
+   default_sigpipe,
+)
+
+class Bootloader(object):
+
+    NAME = None
+    DEFAULT_IMAGE_PATH = '/tmp/sonic_image'
+
+    def get_current_image(self):
+        """returns name of the current image"""
+        with open('/proc/cmdline') as f:
+            current = re.search(r"loop=/*(\S+)/", f.read()).group(1)
+        return current.replace(IMAGE_DIR_PREFIX, IMAGE_PREFIX)
```

Comment:
I understand you created function bodies for get_current_image(), get_binary_image_version(), and verify_binary_image() in this base class because the functionality is shared between GRUB and Uboot. However, I wonder if we should make this class completely abstract and define those functions in the subclasses. Yes, it would mean a small amount of code is duplicated, but it would mean each bootloader is encapsulated. I don't know if it makes sense to have "default" methods here. Thoughts?

Reply:
That was indeed the purpose. Your comment makes sense to me.
What do you feel about creating an `OnieInstallerBootloader` class between `Bootloader` and `(GrubBootloader, UbootBootloader)` ?
In the end the only thing shared by these 2 bootloaders is the format of the installer image.

Reply:
Let me know what you think of this new version.

Reply:
I think this proposal is reasonable, and eliminates the duplicate code via the intermediate `OnieInstallerBootloader` class.